### PR TITLE
Bugfix/Subseasonal EE2 Edits

### PR DESCRIPTION
**Pull request by ShannonShields-NOAA**
## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>
The bugfix for Week 4 and Weeks 3-4 gen_ens_prod.nc file read errors has been tested on the command line.
- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
Checkout feature branch and make sure paths are set for your environment.
Test run GEFS prep via:
EVS/dev/drivers/scripts/subseasonal/prep/jevs_subseasonal_gefs_prep.sh
As always for the subseasonal component, make sure you have 35 days of prep data to run the stats. You can refer to /lfs/h2/emc/vpppg/noscrub/shannon.shields/EVS_Data/evs/v1.0/prep/subseasonal.
Test run the following stats:
EVS/dev/drivers/scripts/subseasonal/stats/jevs_subseasonal_cfs_grid2grid_stats.sh
jevs_subseasonal_gefs_grid2grid_stats.sh
jevs_subseasonal_cfs_grid2obs_stats.sh
Test run a plots job:
EVS/dev/drivers/scripts/subseasonal/plots/jevs_subseasonal_grid2grid_anom_plots_31days.sh
- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**
Yes
- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**
Yes
- [ ] Please complete this pull request review by **[Fill in date]**.</br>
Before Oct. 13
## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

**Comment by AliciaBentley-NOAA** [comment]
@ShannonShields-NOAA Please update the names of your prepbufr dev drivers to be all lowercase again. Thanks!

dev/drivers/scripts/subseasonal/plots/jevs_subseasonal_grid2obs_**PrepBufr**_plots_31days.sh

dev/drivers/scripts/subseasonal/plots/jevs_subseasonal_grid2obs_**PrepBufr**_plots_90days.sh

**Comment by AliciaBentley-NOAA** [comment]
@ShannonShields-NOAA I've started reviewing your subseasonal PR! A few things that I've noticed so far:

1) Steven Earle says that we need to use PDY when setting dates, which means that you need to remove VDATE_START and VDATE_END from your dev drivers. Instead, these can be defined in your J-jobs after PDY is set (which you already do). That said, I noticed something that I wanted to point out here. 

In your 31-day plot dev driver scripts, you set NDAYS=31. You then set "VDATE_START" to -32 days manually (without using NDAYS). For example: 

export NDAYS=31
export VDATE_START=$(date -d "today -32 day" +"%Y%m%d")
export VDATE_END=$(date -d "today -2 day" +"%Y%m%d")

However, when I look at your plotting J-job, I see that you set "VDATE_START" to $PDYm$NDAYS (which I think would be PDYm31). 

(From J-job)
export VDATE_START=${VDATE_START:-$(eval echo \$PDYm$NDAYS)}
export VDATE_END=${VDATE_END:-$PDYm2}

If NCO ran your code without using your dev driver, they'd be using -31 days instead of -32 days. If you really need -32 days for these plots to work correctly, you can make NDAYS=32. The same is true for the 90-day plots, where you set NDAYS=90. but manually set "VDATE_START" to -91 days. 

----------------------------

This is all a vey detailed way of saying that you need to do the following:
1) Remove export VDATE_START, VDATE_END, and COMOUT from your dev drivers. 
2) Make sure that VDATE_START, VDATE_END, and COMOUT are set the way you need them to be in your J-jobs (and maybe thats with

**Comment by AliciaBentley-NOAA** [comment]
Oops! Hit the wrong button while trying to leave a comment! I will try again!

**Comment by AliciaBentley-NOAA** [comment]
@ShannonShields-NOAA I've started reviewing your subseasonal PR! A few things that I've noticed so far:

Steven Earle says that we need to use PDY when setting our dates. You can do this by by removing VDATE_START and VDATE_END from your dev drivers where you don't use PDY. VDATE_START and VDATE_END are already defined in your J-jobs after PDY is set, which is great! That said, I noticed some discrepancies between the values in your dev drivers and J-jobs that I wanted to point out here.

In your 31-day plot dev driver scripts, you set NDAYS=31. You then manually set "VDATE_START" to -32 days (without using NDAYS). For example:

```
export NDAYS=31
export VDATE_START=$(date -d "today -32 day" +"%Y%m%d")
export VDATE_END=$(date -d "today -2 day" +"%Y%m%d")
```

However, when I look at your plotting J-job, I see that you set "VDATE_START" to $PDYm$NDAYS (which I think would be PDYm31).

```
export VDATE_START=${VDATE_START:-$(eval echo \$PDYm$NDAYS)}
export VDATE_END=${VDATE_END:-$PDYm2}
```

If NCO ran your code without using your dev driver, which they will, they'd be using -31 days instead of -32 days! If you need -32 days for these plots to work correctly, you can make NDAYS=32! The same is true for your 90-day plots, where you set NDAYS=90, but manually set "VDATE_START" to -91 days.

-------------

This is all a vey detailed way of saying that you need to do the following:

1) Remove export VDATE_START, VDATE_END, and COMOUT from your dev drivers.
2) Set VDATE_START, VDATE_END, and COMOUT the way that you need them to be in your J-jobs (and maybe thats with NDAYS=32 and NDAYS=91 in your dev drivers.

**Comment by AliciaBentley-NOAA** [comment]
@ShannonShields-NOAA Please change "set -xa" to "set -x" (without the "a") at the beginning of your j-jobs.

**Comment by AliciaBentley-NOAA** [comment]
@ShannonShields-NOAA Please set "export SENDDBN=${SENDDBN:-NO}" in your prep and stats J-jobs. You can keep "export SENDDBN=${SENDDBN:-YES}" in your plots J-jobs, but we don't actually want to send NCO any SENDDBN alerts yet. To avoid this, please add "export SENDDBN=NO" into your **plotting** dev driver scripts. Thanks!

File: modulefiles/subseasonal/subseasonal_plots.sh [hunk 1]
```diff
@@ -3,7 +3,8 @@
 
 set +x
```
Comment (AliciaBentley-NOAA) [inline modulefiles/subseasonal/subseasonal_plots.sh]: @ShannonShields-NOAA Please change this from "set +x" to "set -x"

File: modulefiles/subseasonal/subseasonal_prep.sh [hunk 2]
```diff
@@ -3,7 +3,8 @@
 
 set +x
```
Comment (AliciaBentley-NOAA) [inline modulefiles/subseasonal/subseasonal_prep.sh]: @ShannonShields-NOAA Please change this from "set +x" to "set -x"

File: modulefiles/subseasonal/subseasonal_stats.sh [hunk 3]
```diff
@@ -3,7 +3,8 @@
 
 set +x
```
Comment (AliciaBentley-NOAA) [inline modulefiles/subseasonal/subseasonal_stats.sh]: @ShannonShields-NOAA Please change this from "set +x" to "set -x"

**Comment by AliciaBentley-NOAA** [comment]
@ShannonShields-NOAA I have finished my review of your PR against the EE2 edits document! Please make the changes that I've pointed out above and let us know when your PR is ready for testing!

@PerryShafran-NOAA @ShelleyMelchior-NOAA

**Comment by ShannonShields-NOAA** [comment]
@AliciaBentley-NOAA I have made most of the edits as you can see in the latest commit. However, we have an issue with the PDY settings you suggest. The dates are set just so based on the fact that subseasonal runs are centered around PDYm2, not PDYm1 like most other components. I cannot make the NDAYS change because that needs to remain set to 31 and 90 respectively...otherwise scripts will fail such as for the large tar file, which is set based on NDAYS naming convention. Since I cannot change the NDAYS, I'm hoping I can just change "export VDATE_START=${VDATE_START:-$PDYm32}" in my j-jobs (using 31 day plot example here). Is this acceptable for NCO standards?

**Comment by AliciaBentley-NOAA** [comment]
@ShannonShields-NOAA I think that is acceptable! NCO just wants all dates to be set using PDY (in case ops breaks and NCO needs to backfill by using a previous PDY). As long as you're using PDY and not the "current date" when the job is run, you're fine! 

Try things out after you've 1) removed export VDATE_START, VDATE_END, and COMOUT from your dev drivers and 2) set VDATE_START, VDATE_END, and COMOUT the way that you need them to be set in your J-jobs using PDY (and not using $NDAYS). Thanks for making these changes!

**Comment by ShannonShields-NOAA** [comment]
Great! Thanks @AliciaBentley-NOAA , I will do so and let you know how it goes.

**Comment by ShannonShields-NOAA** [comment]
My mind has been on so many things that I forgot to realize that what I propose for the j-job won't work because there is only one j-job to run 31 and 90 days. So the only other option left is to export a new DAYS variable in dev/drivers to then read in j-job as "export VDATE_START=${VDATE_START:-$PDYm$DAYS}." I have tested this and it works. Does this sound ok @AliciaBentley-NOAA before I implement this change?

**Comment by AliciaBentley-NOAA** [comment]
@ShannonShields-NOAA Yes, that sounds good to me!

**Comment by ShannonShields-NOAA** [comment]
@PerryShafran-NOAA and @ShelleyMelchior-NOAA When you get the chance/after testing previous PRs, this subseasonal bugfix/EE2 PR is ready for testing. For the first test, run GEFS prep via:
EVS/dev/drivers/scripts/subseasonal/prep/jevs_subseasonal_gefs_prep.sh
Given our ongoing WCOSS2 issues, it is highly likely that this job will not finish, but that is ok for our purposes here. I just want to test the relatively simple changes made to the "high-level" dev/drivers and j-jobs.

**Comment by PerryShafran-NOAA** [comment]
OK, I am getting ready for testing now.

**Comment by PerryShafran-NOAA** [comment]
Hi, @ShannonShields-NOAA 

There is a problem.  By removing COMOUT from your driver script, the script now thinks you are writing to the prod directory.  See here: /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr278test/EVS/dev/drivers/scripts/subseasonal/prep/jevs_subseasonal_gefs_prep.o83409751


```
BEGIN: get_gefs_subseasonal_data_files_prep.py
Traceback (most recent call last):
  File "/lfs/h2/emc/vpppg/noscrub/perry.shafran/pr278test/EVS/ush/subseasonal/get_gefs_subseasonal_data_files_prep.py", line 260, in 
<module>
    os.makedirs(dest_model_dir)
  File "/apps/spack/python/3.10.4/intel/19.1.3.304/xqft4d45h4dp4xnbz2ue3nbxv65i6bgp/lib/python3.10/os.py", line 215, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/apps/spack/python/3.10.4/intel/19.1.3.304/xqft4d45h4dp4xnbz2ue3nbxv65i6bgp/lib/python3.10/os.py", line 215, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/apps/spack/python/3.10.4/intel/19.1.3.304/xqft4d45h4dp4xnbz2ue3nbxv65i6bgp/lib/python3.10/os.py", line 215, in makedirs
    makedirs(head, exist_ok=exist_ok)
  [Previous line repeated 2 more times]
  File "/apps/spack/python/3.10.4/intel/19.1.3.304/xqft4d45h4dp4xnbz2ue3nbxv65i6bgp/lib/python3.10/os.py", line 225, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/lfs/h1/ops/prod/com/evs'
```

Here is what COMOUT is set as:

```
0 + export COMOUT=/lfs/h1/ops/prod/com/evs/v1.0/prep/subseasonal/atmos
0 + COMOUT=/lfs/h1/ops/prod/com/evs/v1.0/prep/subseasonal/atmos
```

@AliciaBentley-NOAA Tagging because she suggested to remove COMOUT from the driver.  But apparently it needs to be put back in the driver so we are able to write to COMOUT.

Perry

**Comment by ShannonShields-NOAA** [comment]
@PerryShafran-NOAA This is strange because this is how it is set up in the EVS subseasonal parallel and that works fine:
compath.py -o evs/v1.0/prep/subseasonal/atmos
1 + export COMIN=/lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/prep/subseasonal/atmos
So granted this is for COMIN for a stats job, but the path is the same at the beginning. Looking at this, I assumed that the "compath.py -o" part was /lfs/h2/emc/vpppg/noscrub/$USER/. Is something different in the parallel set-up for this?

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA Shannon can't put her current COMOUT back into her dev/drivers/scripts because it includes a VDATE that isn't specified until PDY is set in the J-job. She can, however, add export COMROOT=/lfs/h2/emc/vpppg/noscrub/$USER back into her dev/drivers/scripts, which should prevent her from trying to write to prod /com. The compath.py command uses COMROOT. 
@ShannonShields-NOAA

**Comment by AliciaBentley-NOAA** [comment]
@ShannonShields-NOAA @PerryShafran-NOAA 
Also of note, I see some hardcoded paths in the /dev/drivers/scripts that I didn't see before. For example:

export MET_ROOT=/apps/ops/prod/libs/intel/${intel_ver}/met/${met_ver}

Can we please remove these hardcoded paths, please?

**Comment by ShannonShields-NOAA** [comment]
Oh so I didn't realize that...my previous statement about compath.py is wrong then. compath.py uses COMROOT so that is why the paths worked before?

**Comment by AliciaBentley-NOAA** [comment]
@ShannonShields-NOAA Correct. I didn't actually ask you to remove COMROOT from your dev/drivers/scripts (I only asked about removing VDATE_START, VDATE_END, and COMOUT), but it was right next to them, so I get why it was lumped into the removal! 

I suspect that if you add "export COMROOT=/lfs/h2/emc/vpppg/noscrub/$USER" back into your dev/drivers/scripts that this might work again!

**Comment by ShannonShields-NOAA** [comment]
@AliciaBentley-NOAA Ok, sorry about that; I will add COMROOT back in.
As for export MET_ROOT, that is needed to be defined for the METplus create_job_scripts.py and util.py scripts. Without this defined, code will break. Now I see Mallory took this out of her scripts, but still has:
METPLUS_PATH = os.environ['METPLUS_PATH']
MET_ROOT = os.environ['MET_ROOT']
in her scripts like I am doing. If she took it out of her dev/drivers, then where is MET_ROOT being defined? This should go for every component not just global_det, so how is everyone defining this?

**Comment by ShannonShields-NOAA** [comment]
Digging deeper, I see machine.conf in has:
MET_INSTALL_DIR = {ENV[MET_ROOT]}
METPLUS_BASE = {ENV[METPLUS_PATH]}
So people must be defining MET_ROOT and METPLUS_PATH somewhere in their own scripts..

**Comment by AliciaBentley-NOAA** [comment]
@ShannonShields-NOAA [A search of MET_ROOT in the EVS auth repo](https://github.com/search?q=repo%3ANOAA-EMC%2FEVS+MET_ROOT&type=code&p=1) reveals that it is only defined using a hardcoded path in subseasonal and in an old nfcens script. 

@malloryprow has probably stopped working for the day, but I'll tag @MarcelCaron-NOAA and @ShelleyMelchior-NOAA so that they can explain to us how people are defining MET_ROOT and METPLUS_BASE without using a hardcoded path. Thanks, all!

**Comment by ShannonShields-NOAA** [comment]
Thank you @AliciaBentley-NOAA ! I appreciate it.

**Comment by ShannonShields-NOAA** [comment]
So @PerryShafran-NOAA I will work on replacing COMROOT for the stats and plots drivers. However, I will need to keep COMOUT as it was defined before for the prep drivers I realize because based on how I understand compath.py now, if I set COMROOT=/lfs/h2/emc/vpppg/noscrub/$USER then compath.py will not point to /lfs/h1/ops/prod/com/. This would then break the path for the model and obs data. I'll let you know when I am done with all of this.

**Comment by MarcelCaron-NOAA** [comment]
> @ShannonShields-NOAA [A search of MET_ROOT in the EVS auth repo](https://github.com/search?q=repo%3ANOAA-EMC%2FEVS+MET_ROOT&type=code&p=1) reveals that it is only defined using a hardcoded path in subseasonal and in an old nfcens script.
> 
> @malloryprow has probably stopped working for the day, but I'll tag @MarcelCaron-NOAA and @ShelleyMelchior-NOAA so that they can explain to us how people are defining MET_ROOT and METPLUS_BASE without using a hardcoded path. Thanks, all!

@ShannonShields-NOAA I believe MET_ROOT and METPLUS_BASE are defined when the met and metplus modules are loaded and don't need to be explicitly defined in the driver script.  They should already exist in the environment.  

You can test this by commenting out the METPLUS_BASE and MET_ROOT definitions and the J-job execution line in one of the subseasonal driver scripts, then sourcing that script and echoing METPLUS_BASE and MET_ROOT on the command line.

**Comment by ShannonShields-NOAA** [comment]
Ah thank you @MarcelCaron-NOAA ! That makes sense given how no one else is defining these variables like I am. I will test this.

**Comment by AliciaBentley-NOAA** [comment]
@MarcelCaron-NOAA Thank you for your help! 

@ShannonShields-NOAA Thanks for looking into COMIN and COMOUT more! JEVS_SUBSEASONAL_PREP looks perfect to me, and JEVS_SUBSEASONAL_STATS and JEVS_SUBSEASONAL_PLOTS are almost perfect. Just a couple small things. 

1) If you export COMIN to point to the emc.vpppg parallel **in your stats dev drivers**, then you can correctly set COMIN to the following in JEVS_SUBSEASONAL_STATS:
`export COMIN=${COMIN:-$(compath.py ${envir}/com/$NET/$evs_ver/prep/$COMPONENT/$RUN)}`
(NCO doesn't want COMIN to be defined using compath.py **-o** in ops; only COMOUT lines should use the -o)

2) If you export COMIN to point to the emc.vpppg parallel **in your plots dev drivers**, then you can correctly set COMIN to the following in JEVS_SUBSEASONAL_PLOTS:
`export COMIN=${COMIN:-$(compath.py ${envir}/com/$NET/$evs_ver)}`
(NCO doesn't want COMIN to be defined using compath.py **-o** in ops; only COMOUT lines should use the **-o**)

The COMIN lines in the stats and plots J-jobs will work when EVSv1 is operational and located in /lfs/h1/ops/prod/com/evs/! We'll override that by defining COMIN in your stats and plots dev drivers for now. @PerryShafran-NOAA can correct me if I'm wrong, but I think this will do it! Thanks!

**Comment by PerryShafran-NOAA** [comment]
@ShannonShields-NOAA @AliciaBentley-NOAA Yes, absolutely.  Please define COMIN in your dev driver scripts in all cases.  That way we can set up the J-job the way NCO wants us to have them.  We override that by defining them in the driver scripts.

**Comment by PerryShafran-NOAA** [comment]
I pulled in all latest changes.  Please let me know when we are ready to test.

Perry

**Comment by ShannonShields-NOAA** [comment]
@PerryShafran-NOAA I have made all of the changes described above. Please recheck out this feature branch to pull in all of the latest changes. Once that is done, please proceed with the GEFS prep test:
dev/drivers/scripts/subseasonal/prep/jevs_subseasonal_gefs_prep.sh

**Comment by PerryShafran-NOAA** [comment]
Prep step is underway!

**Comment by PerryShafran-NOAA** [comment]
Hi, @ShannonShields-NOAA ,

I'm sorry, but the gefs prep job failed.

See /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr278test/EVS/dev/drivers/scripts/subseasonal/prep/jevs_subseasonal_gefs_prep.o83466377

```
BEGIN: get_gefs_subseasonal_data_files_prep.py
Traceback (most recent call last):
  File "/lfs/h2/emc/vpppg/noscrub/perry.shafran/pr278test/EVS/ush/subseasonal/get_gefs_subseasonal_data_files_prep.py", line 232, in <module>
    time_info_dict = get_time_info(
  File "/lfs/h2/emc/vpppg/noscrub/perry.shafran/pr278test/EVS/ush/subseasonal/get_gefs_subseasonal_data_files_prep.py", line 74, in get_time_info
    valid_time = init_time + datetime.timedelta(hours=int(lead))
ValueError: invalid literal for int() with base 10: ''
CFP RANK   0    CFP TASK NUMBER: 0001 FAILED.     USER COMMAND: /lfs/h2/emc/stmp/perry.shafran/evs_test/prod/tmp/jevs_subseasonal_gefs_prep.83466377.cbqs01/prep/prep_job_scripts/retrieve_data/job1
CFP RANK   0    TOTAL RANK RUN TIME:   1.5 sec    Return status: 00000001 hex
nid001094.cactus.wcoss2.ncep.noaa.gov: rank 0 exited with code 1
```

I'm not sure if any of your changes caused this or what.  We'll need to take a look at this.

Perry

**Comment by ShannonShields-NOAA** [comment]
@PerryShafran-NOAA Ok, so I was looking at the output before this message and noticed that your output directory (/lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/prep/subseasonal/atmos.20231002/gefs) already had all of the model data. I think you maybe linked to my own directory (which already finished) before running this test. Therefore, there are no forecast hours for this job to run hence the error we are seeing. So you need to remove the entire atmos.20231002/gefs directory and then please re-test GEFS prep.

**Comment by PerryShafran-NOAA** [comment]
Ah, I *was* linked to your directory, and that could have been a problem.  I unlinked your directory and am trying again.

Perry

**Comment by ShannonShields-NOAA** [comment]
@PerryShafran-NOAA I am happy to report that the GEFS prep job has successfully finished with no errors.
Next, let's test a G2G stats job:
dev/drivers/scripts/subseasonal/stats/jevs_subseasonal_gefs_grid2grid_stats.sh
From what I saw in your prep directory before, I think you have the 35 days of prep data needed to run this.

**Comment by PerryShafran-NOAA** [comment]
For the stats job, I will link to the emc.vpppg or your prep archive and then run the stats job.

**Comment by ShannonShields-NOAA** [comment]
Ugh. Sorry @PerryShafran-NOAA , as you can see in my latest commit, I had a typo that stopped the stats job. Please pull in this recent change before running jevs_subseasonal_gefs_grid2grid_stats.sh again.

**Comment by PerryShafran-NOAA** [comment]
It's running again.

**Comment by PerryShafran-NOAA** [comment]
It looks like all is good this time.

Please see the .o file:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr278test/EVS/dev/drivers/scripts/subseasonal/stats/jevs_subseasonal_gefs_grid2grid_stats.o83477659

And the stat file here: /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/stats/subseasonal/gefs.20231002

**Comment by ShannonShields-NOAA** [comment]
Yes, everything ran as it should. There are some missing averages due to our many machine issues over the past weeks, but the scripts ran correctly and skipped over the periods in which there was not enough model data.
In order to be thorough and to run a stats job with more data, please test CFS: jevs_subseasonal_cfs_grid2grid_stats.sh

**Comment by PerryShafran-NOAA** [comment]
On it!

**Comment by ShannonShields-NOAA** [comment]
@PerryShafran-NOAA Everything looks good and I can see no errors with this CFS test. One more test for G2O before we move on to plots testing:
jevs_subseasonal_cfs_grid2obs_stats.sh

**Comment by PerryShafran-NOAA** [comment]
Thanks for checking!  I appreciate it.  I've been going in so many different directions that past few days.  Thanks for being on the ball!

**Comment by PerryShafran-NOAA** [comment]
@ShannonShields-NOAA The cfs grid2obs stats job is done if you'd like to take a look-see.

**Comment by ShannonShields-NOAA** [comment]
Yes, all good and that concludes the stats testing. For the final test, please run this plots job:
dev/drivers/scripts/subseasonal/plots/jevs_subseasonal_grid2grid_anom_plots_31days.sh

**Comment by ShannonShields-NOAA** [comment]
@PerryShafran-NOAA The plots job finished successfully with the number of available stat files. That concludes all testing and this PR should be ready to merge.

**Review by PerryShafran-NOAA — APPROVED** [review]
Everything looks ready to go!

**Pull request merged**